### PR TITLE
Mobile: Create Mobile V2 User Endpoint

### PR DESCRIPTION
**Pull request by Tonksthebear**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
Mobile has split up the User endpoint into multiple so that information can be requested as needed. This adds the base "User" endpoint

## Related issue(s)
- [department-of-veterans-affairs/va.gov-team#6404](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/6404)

## Testing done
Added specs for the new endpoint

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
None, new endpoint

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/mobile/docs/schemas/v2/User.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    example: "user"
+  attributes:
+    type: object
+    required:
+      - id
+      - firstName
+      - middleName
+      - lastName
+      - signinEmail
+      - signinService
```
Comment (aherzberg) [inline modules/mobile/docs/schemas/v2/User.yml:21]: isCernerPatient looks to be missing from required.

Comment (jperk51) [reply modules/mobile/docs/schemas/v2/User.yml:21]: @kpethtel is this something that you think should be part of the facilities endpoint instead of the user endpoint?

**Review by kpethtel — COMMENTED** [review]
The user show code looks fine, but related to a conversation we had here: https://github.com/department-of-veterans-affairs/vets-api/pull/13481#discussion_r1290632040

I was expecting we'd also carry over the before actions from the older versions of the user show. That includes precaching resources and the vets 360 linking. Alternatively, it could be worth discussing alternative ways of handling those. We're using the user show as a proxy for login, but that's not ideal IMO, in part because the front end sometimes calls it for other reasons. It could be worth asking the SIS team if there's a way for us to add some actions after a mobile user logs in. Or we could just create some sort of login endpoint that doesn't return data but exists just to execute login actions.

File: modules/mobile/docs/schemas/v2/User.yml (hunk 1 above)
Comment (kpethtel) [reply modules/mobile/docs/schemas/v2/User.yml:21]: Facilities info does already have cerner field and the front end uses it for `userHasCernerFacilities`. They use that method instead of isCernerPatient, afaict. I'll ask them if they think it's worth getting rid of, but it also doesn't seem to have any carrying cost.

Comment (jperk51) [reply modules/mobile/docs/schemas/v2/User.yml:21]: Yeah, as long as it doesn't require calling an additional upstream then it's whatever

Comment (kpethtel) [reply modules/mobile/docs/schemas/v2/User.yml:21]: The user instantiation code is very difficult to follow, but I believe it likely does make a request to MPI for this data if that data isn't already cached. Given that this is the user endpoint and probably the first thing the mobile app fetches, it may not be cached. The code is so hard to follow that I can't be totally sure, especially given that the vets-api pre-loads some (but not all) data and caches some (but maybe not all) data. 

I've asked the FE team about this. If they say it's not necessary, we can remove it. But I also think we should generally be able to call most methods on the user object without concern.

**Comment by Tonksthebear** [comment]
I'm good with just adding back those before_actions unless you think it's important that this code lives elsewhere

File: modules/mobile/docs/schemas/v2/User.yml (hunk 1 above)
Comment (kpethtel) [reply modules/mobile/docs/schemas/v2/User.yml:21]: No one has answered my question, so I'm just going to say I think we can safely remove it. If front end wants it back, they can ask for it back when they implement the front end of this ticket. It does seem like accessing the info can make an upstream request so we should avoid calling it if we don't need to.

Comment (kpethtel) [reply modules/mobile/docs/schemas/v2/User.yml:21]: In the specs, it may be handled by the factories. But I doubt this should be a big issue either way. It's also possible that the user gets fully loaded by the authentication process and it's just not clear from the code. Very hard to say.

**Comment by kpethtel** [comment]
> I'm good with just adding back those before_actions unless you think it's important that this code lives elsewhere

@Tonksthebear I vote for writing another ticket for this. It needs to go somewhere, whether that's here, another endpoint, or as part of the login process. I think we can put it in a ticket and look at our options.

File: modules/mobile/docs/schemas/v2/User.yml (hunk 1 above)
Comment (Tonksthebear) [reply modules/mobile/docs/schemas/v2/User.yml:21]: Removed!

**Review by kpethtel — APPROVED** [review]
This does the vet360 linking but not the resource pre-loading, so it feels like a half-measure. I'm fine with that on the condition that we file a ticket to reconsider how we want it done. 

As for the cerner patient attribute, I defer to Jayson on that, but I think we also do have to rely on the user methods available to us to some extent on the assumption that they'll be fast and stable. It could be worth talking to Trevor on the SIS team about, since I know he has experience in the user instantiation process.

**Comment by Tonksthebear** [comment]
Added a ticket

**Review by aherzberg — APPROVED** [review]
LGTM
